tests/openfl/federated/data/test_verifiable_dataset.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from openfl.federated.data.sources.local_data_source import LocalDataSource
+from openfl.federated.data.sources.verifiable_dataset_info import VerifiableDatasetInfo
+import pytest
+import os
+
+from pathlib import Path
+from typing import List, Tuple
+
+@pytest.fixture
+def local_data_sources(fs) -> Tuple[Path, Path]:
+    """Fixture to create two data sources with a file tree structure using pyfakefs."""
+    base_tmp = Path("/test_data")  # Fake base path
+
+    # Define datasource paths
+    ds1 = base_tmp / "datasource1"
+    ds2 = base_tmp / "datasource2"
+
+    for ds in [ds1, ds2]:
+        fs.create_dir(ds / "1")
+        fs.create_dir(ds / "2")
+
+        for subdir in ["1", "2"]:
+            for i in range(1, 4):
+                file_path = ds / subdir / f"file{i}.txt"
+                fs.create_file(file_path, contents=f"Hello world! {ds.name} dir {subdir} file{i}\n")
+
+    return ds1, ds2  # Return fake paths
+
+def split_to_base_and_relative_paths(data_sources_paths: List[str]) -> Tuple[str, List[str]]:
+    """Split a list of paths into a base directory and relative paths."""
+    absolute_paths = [os.path.realpath(path) for path in data_sources_paths]
+    base_path = os.path.commonpath(absolute_paths)
+    relative_paths = [os.path.relpath(path, base_path) for path in absolute_paths]
+    return base_path, relative_paths
+
+def copy_subtree(fs, existing_dir_path, new_dir_tree):
+    """Recursively copy all files and directories from existing_dir_path to new_dir_tree in pyfakefs."""
+    fs.create_dir(new_dir_tree)  # Ensure the new directory exists
+
+    for subpath in existing_dir_path.iterdir():
+        new_path = new_dir_tree / subpath.name
+        if subpath.is_dir():
+            copy_subtree(fs, subpath, new_path)  # Recursively copy subdirectories
+        else:
+            file_content = fs.get_object(subpath).contents  # Read from fake filesystem
+            fs.create_file(new_path, contents=file_content)  # Create file in new location
+
+def test_one_local_datasource(local_data_sources):
+    ds1, _ = local_data_sources
+    base_path, relative_paths = split_to_base_and_relative_paths([ds1])
+    datasources = [LocalDataSource(source_path=rel_path, base_path=base_path) for rel_path in relative_paths]
+    verifiable = VerifiableDatasetInfo(data_sources=datasources, label="my_dataset", metadata="md")
+    hash = verifiable.create_dataset_hash()
+    assert isinstance(hash, str), f"Expected str, got {type(hash)}"
+    verifaible_json = verifiable.to_json()
```

Comment:
Minor : Typo : verifaible_json 
```suggestion
    verifiable_json = verifiable.to_json()
```